subprojects/logging/src/main/java/org/gradle/internal/deprecation/DeprecationMessageBuilder.java

```diff
@@ -285,6 +285,39 @@ String formatAdvice(String replacement) {
         }
     }
 
+    public static class DeprecateSystemProperty extends WithReplacement<String, DeprecateSystemProperty> {
+        private final String systemProperty;
+
+        DeprecateSystemProperty(String systemProperty) {
+            super(systemProperty);
```

Comment:
You should add a call to `setIndirectUsage()` here, since the code emitting the deprecation message never will be the user code setting the system property. This avoids that the deprecation has a stacktrace which points to some Gradle internal code, see e.g. here: https://ge.gradle.org/s/rhnlfxe7drcda/deprecations?expanded=WyI0Il0#5

Reply:
Huh, good to know. I wonder how to test this properly then.

Reply:
Created https://github.com/gradle/gradle/pull/14342.